charts/k8s-monitoring/charts/feature-pod-logs/templates/_filelog.alloy.tpl

```diff
@@ -0,0 +1,158 @@
+  exclude = [
+{{- range .Values.excludeNamespaces }}
+    "/var/log/pods/{{ . }}_*/*/*.log",
+{{- end }}
+  ]
+{{- end }}
+  start_at = {{ if .Values.filelogGatherSettings.onlyGatherNewLogLines }}"end"{{ else }}"beginning"{{ end }}
+  include_file_name = false
+  include_file_path = true
+
+  operators = [
+    // Container operator will set k8s.pod.name, k8s.pod.uid, k8s.container.name, k8s.container.restart_count, and k8s.namespace.name
+    {
+      type                       = "container",
+      add_metadata_from_filepath = true,
+    },
+  ]
+
+  output {
+    logs = [otelcol.processor.k8sattributes.pod_logs.input]
+  }
+}
+
+otelcol.processor.k8sattributes "pod_logs" {
+  pod_association {
+    source {
+      from = "resource_attribute"
+      name = "k8s.pod.uid"
+    }
+  }
+
+  extract {
+    metadata = [
+      "k8s.deployment.name",
+      "k8s.statefulset.name",
+      "k8s.daemonset.name",
+      "k8s.cronjob.name",
+      "k8s.job.name",
+      "k8s.node.name",
+    ]
+    annotation {
+      key_regex = "(.*)"
+      tag_name  = "$1"
+      from      = "pod"
+    }
+    annotation {
+      key_regex = "resource.opentelemetry.io/(.*)"
+      tag_name  = "$1"
+      from      = "pod"
+    }
+{{- range $attribute, $annotation := .Values.annotations }}
+    annotation {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $annotation | quote }}
+      from     = "pod"
+    }
+{{- end }}
+{{- range $attribute, $annotation := .Values.nodeAnnotations }}
+    annotation {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $annotation | quote }}
+      from     = "node"
+    }
+{{- end }}
+{{- range $attribute, $annotation := .Values.namespaceAnnotations }}
+    annotation {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $annotation | quote }}
+      from     = "namespace"
+    }
+{{- end }}
+    label {
+      key_regex = "(.*)"
+      tag_name  = "$1"
+      from      = "pod"
+    }
+{{- range $attribute, $label := .Values.labels }}
+    label {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $label | quote }}
+      from     = "pod"
+    }
+{{- end }}
+{{- range $attribute, $label := .Values.nodeLabels }}
+    label {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $label | quote }}
+      from     = "node"
+    }
+{{- end }}
+{{- range $attribute, $label := .Values.namespaceLabels }}
+    label {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $label | quote }}
+      from     = "namespace"
+    }
+{{- end }}
+  }
+
+  output {
+    logs = [otelcol.processor.transform.pod_logs.input]
+  }
+}
+
+otelcol.processor.transform "pod_logs" {
+  error_mode = "ignore"
+  log_statements {
+    context = "resource"
+    statements = [
+      `delete_key(attributes, "k8s.container.restart_count")`,
+      `delete_key(attributes, "log.file.path")`,
+
+      `set(attributes["service.name"], attributes["service.name"]) where attributes["service.name"] == nil`,
```

Comment:
```suggestion
```

Reply:
I believe this will already be set by the k8s attribute processor if `resource.opentelemetry.io/service.name` is set.

Reply:
Are we sure about that? I'm not seeing that in the [alloy component docs](https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.processor.k8sattributes/) or the [upstream docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor).

Reply:
yes, I was just noting if the `resource.opentelemetry.io/service.name` annotation is on the pod in question, then the config we have in the `k8satttributesprocessor` step would have set `service.name` resource attribute:

```
    annotation {
      key_regex = "resource.opentelemetry.io/(.*)"
      tag_name  = "$1"
      from      = "pod"
    }
```